tests/typescript/conformance/types/intersectionType/intersectionType.ts

```diff
@@ -0,0 +1,6 @@
+type A = & string;
+type B =
+  & { foo: string }
+  & { bar: number };
+
+type C = [& { foo: 1 } & { bar: 2 }, & { foo: 3 } & { bar: 4 }];
```

Comment:
Yeah, it's allowed with flow as well. We need to make sure that we can parse it, but we should print it in a different way :)